src/Features/CSharp/Portable/Completion/CompletionProviders/ImportCompletion/ImportCompletionProviderHelper.cs

```diff
@@ -21,10 +22,12 @@ internal static class ImportCompletionProviderHelper
             => semanticModel.GetUsingNamespacesInScope(location)
                 .SelectAsArray(namespaceSymbol => namespaceSymbol.ToDisplayString(SymbolDisplayFormats.NameFormat));
 
-        public static async Task<SyntaxContext> CreateContextAsync(Document document, int position, CancellationToken cancellationToken)
+        public static async Task<SyntaxContext?> CreateContextAsync(Document document, int position, CancellationToken cancellationToken)
         {
             // Need regular semantic model because we will use it to get imported namespace symbols. Otherwise we will try to 
             // reach outside of the span and ended up with "node not within syntax tree" error from the speculative model.
+            var sourceText = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            document = sourceText.GetDocumentWithFrozenPartialSemantics(cancellationToken) ?? document;
```

Comment:
1. should comment what we're doing this for.
2. `?? doc` should be unnecessary.  we should throw if this fails.